src/main/java/com/google/roomies/database/Database.java

```diff
@@ -0,0 +1,76 @@
+package com.google.roomies;
+
+import com.google.cloud.firestore.DocumentSnapshot;
+import com.google.cloud.firestore.Firestore;
+import com.google.cloud.firestore.QueryDocumentSnapshot;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map; 
+
+interface Database {
+  /**
+  * Gets the database instance.
+  *
+  * @throws IOException if database cannot be initialized
+  */
+  public Firestore getDatabase() throws IOException;
+
+  /**
+  * Sets the database instance.
+  * Used to set database to a mock for testing.
+  *
+  * @throws IOException if database cannot be initialized
+  */
+  public void setDatabaseForTest(Firestore database);
+  
+  /**
+  * Add a document to a collection using a map.
+  *
+  * @param collectionName name of collection in database
+  * @param doc document that implements the document interface
+  */  
+  public void addDocumentAsMap(String collectionName, Document doc);
+
```

Comment:
I think I'm going to go with the current setup for now, I think this is a good idea though and will implement if there is time later. Since all documents get added in the same way with a map, I think a general function will be ok

Reply:
Okay - keep in mind though, change implementations is easy, changing interfaces is hard. Approving as-is to unblock progress.